### ISSUE TITLE
Q: Does SecondLevelModel handle Group factors?

### ISSUE THREAD

**Issue by chrplr** [nistats/nistats#123]
If I am not mistaken,  the SecondLevelModel can only currently generate one-way anova within-subject models (2 additive factors, subject being one of them). 

If I am correct, it would be good to extend it to allow more factors, notably Group (=between subjects) factor (See SPM flexible factorial design)

Or rename it OneWay-withinSubject.

**Comment by bthirion** [comment]
No, you can specifiy subject-specific covariates, see e.g. in the plot_second_level_design_matrix.py example, the `extra_info_subjects` variable.

**Comment by chrplr** [comment]
But are subject-specific covariates enough? 
If you want to test for interactions between Group (between Subject) and Treatments (within Subject), don't you need different blocks for each group in the design matrix ?

**Comment by chrplr** [comment]
I realize that one reason to use complex factorial models in group analysis in SPM is linked to a limitation of this software. For example, one can only do conjunctions of contrasts in a single model, or extract and plot contrasts values in multiple conditions.

In a way, it is safer and more straightforward to compute within subject contrasts at the first level and to keep the second level analyses simple (e.g, one sample t-test, two sample test, regression) so that, for example, one can assume independence because there are not repeated measures (handled by REML in SPM).

So nistats already fits the bill.

**Comment by bthirion** [comment]
Thx. My opinion is that we do not meet all the possible use cases (e.g. longitudinal and treatment effects), but unless there is a strong pressure use cases (and typically a public dataset to showcase the feature), I am not willing to add more complex features. Not that user-supplied design matrices are still possible.

**Comment by bthirion** [comment]
@chrplr if there is a dataset on hich we can work to improve these features, i'm happy to help. Let me know.
